frontend/src/components/Dashboard.jsx

```diff
@@ -86,546 +86,580 @@ const DashBoardPage = styled.div`
-        );
-        const txHash = await window.ethereum.request({
-          method: "eth_sendTransaction",
-          params: [
-            {
-              from: user.address, // The user's active address.
-              to: PaymasterContract.address, // Required except during contract publications.
-              data: encodedData,
-              value: ethers.utils.parseEther(amount.toString()).toHexString(),
-            },
-          ],
-        });
-        toast.loading(
-          (t) => (
-            <TransactionSentToast
-              txHash={txHash}
-              t={t}
-              blockExplorerLink={networks[chainId].blockExplorerLink}
-            />
-          ),
-          {
-            icon: "👏",
-          }
-        );
-      }
-      setIsLoading(false);
-    } catch (e) {
-      console.error(e.message);
-      toast.error("Something went wrong on MetaMask");
-      setIsLoading(false);
-    }
-  };
+	const handleSubmit = async () => {
+		try {
+			setIsLoading(true);
+			if (!user?.address) {
+				const retUser = await signIn(logInType);
+				if (!retUser) {
+					toast.error("Please make sure that metamask is installed");
+				} else toast.success("Logged in Successfully");
+				return;
+			}
+			if (amount === 0) {
+				toast.error("Please enter an amount to be Deposited/Withdrawn");
+			}
+			const PaymasterContract = getPaymasterContract(chainId);
+			if (checked) {
+				const encodedData = PaymasterContract.interface.encodeFunctionData(
+					"withdrawFunds",
+					[ethers.utils.parseEther(amount.toString())]
+				);
+				const txHash = await window.ethereum.request({
+					method: "eth_sendTransaction",
+					params: [
+						{
+							from: user.address, // The user's active address.
+							to: PaymasterContract.address, // Required except during contract publications.
+							data: encodedData,
+						},
+					],
+				});
+				toast.loading(
+					(t) => (
+						<TransactionSentToast
+							txHash={txHash}
+							t={t}
+							blockExplorerLink={networks[chainId].blockExplorerLink}
+						/>
+					),
+					{
+						icon: "👏",
+					}
+				);
+			} else {
+				const encodedData = PaymasterContract.interface.encodeFunctionData(
+					"depositFunds",
+					[]
+				);
+				const txHash = await window.ethereum.request({
+					method: "eth_sendTransaction",
+					params: [
+						{
+							from: user.address, // The user's active address.
+							to: PaymasterContract.address, // Required except during contract publications.
+							data: encodedData,
+							value: ethers.utils.parseEther(amount.toString()).toHexString(),
+						},
+					],
+				});
+				toast.loading(
+					(t) => (
+						<TransactionSentToast
+							txHash={txHash}
+							t={t}
+							blockExplorerLink={networks[chainId].blockExplorerLink}
+						/>
+					),
+					{
+						icon: "👏",
+					}
+				);
+			}
+			setIsLoading(false);
+		} catch (e) {
+			console.error(e.message);
+			toast.error("Something went wrong on MetaMask");
```

Comment:
Don't assume Metamask, it could be something else

Reply:
for now its just metamask